state/machine.go

```diff
@@ -1241,6 +1242,25 @@ func (m *Machine) InstanceId() (instance.Id, error) {
 	return instData.InstanceId, err
 }
 
+// DisplayName returns the provider-specific display name for this
+// machine, falling back to the instance id if it does not exist.
+func (m *Machine) DisplayName() (instance.DisplayName, error) {
```

Comment:
Perhaps a better way to do this would be to simply return `""`, rather than jumping through hoops to provide a fallback option?

Reply:
You could consider a new method which returns both instance.Id and display name in a struct. I think that could simplify things a fair bit.